torch/testing/_internal/common_methods_invocations.py

```diff
@@ -3716,30 +3800,59 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
            op=torch.diff,
            dtypes=all_types_and_complex_and(torch.bool, torch.float16, torch.bfloat16),
            sample_inputs_func=sample_inputs_diff),
-    OpInfo('div',
-           variant_test_name='no_rounding_mode',
-           dtypes=all_types_and_complex_and(torch.bool, torch.half, torch.bfloat16),
-           sample_inputs_func=sample_inputs_div,
-           skips=(SkipInfo('TestOpInfo', 'test_duplicate_method_tests'),),
-           assert_autodiffed=True),
-    OpInfo('div',
-           variant_test_name='true_rounding',
-           dtypes=all_types_and_complex_and(torch.bool, torch.half, torch.bfloat16),
-           sample_inputs_func=partial(sample_inputs_div, rounding_mode=None),
-           skips=(SkipInfo('TestOpInfo', 'test_duplicate_method_tests'),),
-           assert_autodiffed=True),
-    OpInfo('div',
-           variant_test_name='trunc_rounding',
-           dtypes=all_types_and(torch.half, torch.bfloat16),
-           sample_inputs_func=partial(sample_inputs_div, rounding_mode='trunc'),
-           skips=(SkipInfo('TestOpInfo', 'test_duplicate_method_tests'),),
-           assert_autodiffed=True),
-    OpInfo('div',
-           variant_test_name='floor_rounding',
-           dtypes=all_types_and(torch.half, torch.bfloat16),
-           sample_inputs_func=partial(sample_inputs_div, rounding_mode='floor'),
-           skips=(SkipInfo('TestOpInfo', 'test_duplicate_method_tests'),),
-           assert_autodiffed=True),
+    BinaryUfuncInfo(
+        "div",
+        variant_test_name="no_rounding_mode",
+        aliases=("divide",),
+        ref=np.divide,
+        # prevent zero division
+        domain=(1.0, None),
+        dtypes=all_types_and_complex_and(torch.bool, torch.half, torch.bfloat16),
+        supports_out=True,
+    ),
+    BinaryUfuncInfo(
+        "div",
+        variant_test_name="true_rounding",
+        aliases=("divide",),
+        ref=lambda input, other, **kwargs: np.true_divide(input, other),
+        # prevent zero division
+        domain=(1.0, None),
+        sample_inputs_func=partial(sample_inputs_binary, rounding_mode=None),
+        dtypes=all_types_and_complex_and(torch.bool, torch.half, torch.bfloat16),
+        supports_out=True,
+        skips=(
+            SkipInfo("TestCommon", "test_variant_consistency_jit", dtypes=[torch.float32, torch.complex64]),
+        )
```

Comment:
For some reason this test seem to fail if the `rounding_mode` is explicitly set or `floor_divide` (L410) or `true_divide` (L5521) are used directly. They all fail because the exported / imported dtype does not match the original. This happens for `float32` for all of them and for `complex64` for this OpInfo only.

Since all our dtypes pass, I'm guessing this is not a test error but a legitimate failure.